src/app/shared/components/button/button.component.ts

```diff
@@ -0,0 +1,21 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss']
+})
+export class ButtonComponent implements OnInit {
+
+  @Output()
+  add = new EventEmitter();
+  updateMeals() {
```

Comment:
A shared button component should not have specific functionality like that. It would be better to have it throw an event whenever the user clicks. Then in the component where you'll be using the button you can catch that event and react with a specific function.

So I would remove this.